src/toil/lib/aws/utils.py

```diff
@@ -346,7 +346,7 @@ def bucket_location_to_region(location: Optional[str]) -> str:
     return "us-east-1" if location == "" or location is None else location
 
 
-def get_object_for_url(url: ParseResult, existing: Optional[bool] = None) -> "S3Object":
+def get_object_for_url(url: Union[ParseResult, str], existing: Optional[bool] = None) -> "S3Object":
```

Comment:
I feel like we should have stronger opinions on which of these we ought to have. It's kind of convenient for the caller to just be able to throw whatever in here, but remembering that you *can* do that is harder than remembering that you need one type, I think.

And we're not updating the docstring to say that the URL doesn't *actually* have to be parsed.